### PR TITLE
Alleviation for static analyzers

### PR DESCRIPTION
**Pull request by mmichal10**

### CODE REVIEW

**Review by jfckm — DISMISSED** [review]

File: src/cleaning/alru.c [hunk 1]
```diff
@@ -517,7 +517,7 @@ static int ocf_alru_recovery_handle(ocf_parallelize_t parallelize,
 	int i;
 
 	portion = DIV_ROUND_UP((uint64_t)entries, shards_cnt);
-	begin = portion*shard_id;
+	begin = (uint64_t)portion*shard_id;
 	end = OCF_MIN(portion*(shard_id + 1), entries);
```
Comment (arutk) [inline src/cleaning/alru.c:521]: here such cast is actually required to avoid overflow in portion * (shard_id + 1)

File: src/cleaning/alru.c [hunk 2]
```diff
@@ -517,7 +517,7 @@ static int ocf_alru_recovery_handle(ocf_parallelize_t parallelize,
 	int i;
 
 	portion = DIV_ROUND_UP((uint64_t)entries, shards_cnt);
-	begin = portion*shard_id;
+	begin = (uint64_t)portion*shard_id;
```
Comment (arutk) [inline src/cleaning/alru.c:520]: I would approach it differently. I'm not sure why begin and end are 64 bit in the first place. I would change then to 32bit unsigned.

**Review by arutk — DISMISSED** [review]

**Comment by arutk** [comment]
Sorry, I meant to request changes rather than approve

File: src/cleaning/acp.c [hunk 3]
```diff
@@ -405,7 +405,7 @@ static void ocf_acp_recovery_chunk(struct ocf_acp_recovery_context *context,
 				.chunk[chunk->core_id][chunk->chunk_id];
 	}
 
-	for (bucket_id = 0; bucket_id < ACP_MAX_BUCKETS; bucket_id++) {
```
Comment (arutk) [inline src/cleaning/acp.c:408]: ignore this comment

**Review by arutk — CHANGES_REQUESTED** [review]

**Review by robertbaldyga — APPROVED** [review]

**Pull request merged**